File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        List<char> letters = new List<char>();
        int maxLength = 0;

        foreach(char c in s){

            if(!letters.Contains(c))
                letters.Add(c);

            else{
                maxLength = Math.Max(maxLength, letters.Count);

                int index = letters.IndexOf(c);
                letters.RemoveRange(0, index+1);

                letters.Add(c);
            }
        }

        return Math.Max(maxLength, letters.Count);
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
  
        set<char> c;      
        int res = 0;        
        int l = 0;
        
        for(int r=0; r<s.size(); r++) {
            while(c.find(s[r]) != c.end()) {
                c.erase(s[l]);
                l++;
            }
            c.insert(s[r]);
            
            res = max(res, r-l+1); 
        }

        
        return res;
        
    }
};